heron/schedulers/src/java/com/twitter/heron/scheduler/local/LocalScheduler.java

```diff
@@ -261,12 +262,15 @@ public boolean onUpdate(Scheduler.UpdateTopologyRequest request) {
   @Override
   public void addContainers(Set<PackingPlan.ContainerPlan> containers) {
     synchronized (processToContainer) {
+      Set<Integer> containerIds = new HashSet<>();
+      containerIds.addAll(processToContainer.values());
       for (PackingPlan.ContainerPlan container : containers) {
-        if (processToContainer.containsKey(container.getId())) {
+        if (containerIds.contains(container.getId())) {
```

Comment:
Great catch! Instead of maintaining another Set could you just check `processToContainer.values().containsKey(container.getId())`? That guarantees that we're always in sync with what `startExecutor` has started, since it updates `processToContainer`.

A;ls